mqtt/mqtt-broker/src/broker.rs

```diff
@@ -807,23 +808,24 @@ where
             );
 
             let client_id = connreq.client_id().clone();
-            let (state, client_info_, _will, handle) = current_connected.into_parts();
-            let old_session =
-                Session::new_disconnecting(client_id.clone(), client_info_, None, handle);
-            let (new_session, session_present) =
-                if let proto::ClientId::IdWithExistingSession(_) = connreq.connect().client_id {
-                    debug!(
-                        "moving persistent session to this connection for {}",
-                        client_id
-                    );
-                    let new_session = Session::new_persistent(auth_id, connreq, state);
-                    (new_session, true)
-                } else {
-                    info!("cleaning session for {}", client_id);
-                    let state = SessionState::new(client_id.clone(), self.config.session().clone());
-                    let new_session = Session::new_transient(auth_id, connreq, state);
-                    (new_session, false)
-                };
+            let (state, _will, handle) = current_connected.into_parts();
```

Comment:
On this line, we take the whole state with client_info and pass it for a connecting client which may have different addresses and credentials. I think we should create a new ClientInfo for the connecting client and then update state.

Since we didn't catch it with unit tests, we should add it

Reply:
Scenarion:
1: client connects `{ client_id: "foo", "username": "foo", addr: 198.0.0.1 }`
state has session with client_info { client_id: "foo", "username": "**foo**", addr: **198.0.0.1** }
2: client connects `{ client_id: "foo", "username": "bar", addr: 198.0.0.2 }`
expected: 
state has session with client_info { client_id: "foor", "username": "**bar**", addr: **198.0.0.2** }
actual: 
state has session with client_info { client_id: "foor", "username": "**foo**", addr: **198.0.0.1** }